core/src/main/java/org/aerogear/mobile/core/reactive/CacheRequest.java

```diff
@@ -0,0 +1,78 @@
+package org.aerogear.mobile.core.reactive;
+
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.aerogear.mobile.core.Request;
+import org.aerogear.mobile.core.Responder;
+
+/**
+ * This class wraps a request and subscribes itself. When the request returns a value this request
+ * will save the value and emit it to any responders that attach themselves.
+ * 
+ * @param <T> The result type of the underlying request
+ */
+public final class CacheRequest<T> extends AbstractRequest<T> implements Responder<T> {
+    private final Request<T> delegateTo;
+
+    private T cachedResult;
+    private Exception cachedException;
+
+    private List<AtomicReference<Responder<T>>> awaitingResponders =
+                    Collections.synchronizedList(new ArrayList<>());
+
+    public CacheRequest(Request<T> delegateTo) {
+        nonNull(delegateTo, "delegateTo");
+        this.delegateTo = delegateTo;
```

Comment:
this can be changed to `this.delegateTo = nonNull(delegateTo, "delegateTo")`

Reply:
See https://github.com/aerogear/aerogear-android-sdk/pull/160

Reply:
![image](https://user-images.githubusercontent.com/2413816/37558169-447fddf0-29e6-11e8-8e4f-07b648b43c80.png)